spec/index.html

```diff
@@ -6355,23 +6553,106 @@ <h5>STRDT</h5>
                 </tbody>
               </table>
             </div>
+            <div class="note">
+                `STRDT` should not be called with datatype argument
+                `rdf:dirString` or `rdf:dirLangString. Function `STRLANG`
+                or `STRLANGDIR` should be used.
+            </div>
           </section>
+
           <section id="func-strlang">
             <h5>STRLANG</h5>
             <pre class="prototype nohighlight"><span class="return">literal</span>  <span class="operator">STRLANG</span>(<span class="type">xsd:string</span> lexicalForm, <span class="type">xsd:string</span> langTag)</pre>
-            <p>The <code>STRLANG</code> function constructs a literal with lexical form and language
-              tag as specified by the arguments.</p>
+            <p>The <code>STRLANG</code> function constructs a literal with
+              <a data-cite="RDF12-CONCEPTS#dfn-lexical-form">lexical form</a>
+              and
+              <a data-cite="RDF12-CONCEPTS#dfn-language-tag">language tag</a>
+              as specified by the arguments.
+              as specified by the arguments.
+            </p>
+            <p>
+              The argument `langTag` must not be an empty string.
```

Comment:
Should we state it must (or should?) be a valid [language tag](https://www.w3.org/TR/rdf11-concepts/#dfn-language-tag)

Reply:
I think "should" is better because RDF does not we don't insist on valid language tags - and also "valid" is a potentially moving target.

Change made and also in `STRDIRLANG`.

Also - using MUST and SHOULD.